src/robusta/core/reporting/blocks.py

```diff
@@ -366,3 +367,22 @@ def __init__(self, choices: Dict[str, CallbackChoice]):
         :param choices: a dict mapping between each the text on each button to the action it triggers
         """
         super().__init__(choices=choices)
+
+
+class PrometheusBlock(BaseBlock):
+    """
+    Formatted prometheus query results with metadata
+    """
+
+    data: PrometheusQueryResult
+    metadata: Dict[str, str]
+
+    def __init__(self, data: PrometheusQueryResult, query: str = None):
```

Comment:
I would make the query mandatory, and remove the if below
I think it is mandatory anyway